src/hooks.ts

```diff
@@ -1,21 +1,25 @@
 import { promisify } from 'util';
+import { FinalizePackageTargetsHookFunction, HookFunction } from './types';
 
-export async function promisifyHooks(hooks, args) {
+export async function promisifyHooks(hooks: HookFunction[] | FinalizePackageTargetsHookFunction[] | undefined, args: unknown[]) {
   if (!hooks || !Array.isArray(hooks)) {
     return Promise.resolve();
   }
 
-  await Promise.all(hooks.map(hookFn => promisify(hookFn).apply(this, args)));
+  await Promise.all(hooks.map(hookFn => promisify(hookFn).apply(promisifyHooks, args)));
 }
 
-export function serialHooks(hooks) {
-  return async function() {
-    const args = Array.prototype.splice.call(arguments, 0, arguments.length - 1);
-    const done = arguments[arguments.length - 1];
-    for (const hook of hooks) {
-      await hook.apply(this, args);
-    }
-
-    return done(); // eslint-disable-line promise/no-callback-in-promise
-  };
-}
+/**
```

Comment:
You used to be able to run `const { serialHooks } = require('electron-packager/hooks')`. I'm not sure if we actually need that though, people could just implement it themselves?

Reply:
Getting rid of it would require a major version bump. I literally added it because someone complained about the behavior breaking, so it's probably used in at least one app. It's sounding like we're going to try to bring this PR in as a minor version bump, so we should probably keep it for now (and label it as deprecated?)

Reply:
Did some digging and found the use case for this: https://github.com/electron/electron-packager/commit/a2f3a89e55ac2da639de325856a6924a80c896b1#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abR44-R68. The contents of `docs/api.md` were eventually replaced by a link to https://electron.github.io/electron-packager, so we lost the only mention of this function we had in the project.

That makes me thing we should keep it and just add a comment with that missing docs. We could also make it possible to import it from the top-level package (`import { serialHooks } from 'electron-packager'`) and deprecate / add a deprecation notice for `electron-packager/hooks` (but not for the function itself) if that makes sense.

Reply:
https://github.com/electron/packager/pull/1593/commits/47e4006c666e72aea8347bc98c19db38bb826d28